From bc06c1174f2ca1109c31e00a92d7250d08d401de Mon Sep 17 00:00:00 2001
From: Haihua Hu <jared.hu@nxp.com>
Date: Wed, 7 Dec 2016 16:51:27 +0800
Subject: [PATCH 24/24] gl/effects: use non-PBO GLMemory for internal textures

middle textures in gleffects do not need to use GstGLMemoryPBO as they
aren't transfering data to/from the GPU. This will cost too much DMA
memory and cause performance issue. Change the allocator to use non-PBO
GstGLMemory.

Upstreame-Status: Backport[1.10.3]

https://bugzilla.gnome.org/show_bug.cgi?id=776072

Signed-off-by: Haihua Hu <jared.hu@nxp.com>
---
 ext/gl/gstgleffects.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ext/gl/gstgleffects.c b/ext/gl/gstgleffects.c
index 458d95a..cbb2d1c 100644
--- a/ext/gl/gstgleffects.c
+++ b/ext/gl/gstgleffects.c
@@ -279,7 +279,7 @@ gst_gl_effects_init_gl_resources (GstGLFilter * filter)
   gint i;
 
   base_alloc = (GstGLBaseMemoryAllocator *)
-      gst_gl_memory_allocator_get_default (context);
+      gst_allocator_find (GST_GL_MEMORY_ALLOCATOR_NAME);
   params =
       (GstGLAllocationParams *) gst_gl_video_allocation_params_new (context,
       NULL, &filter->out_info, 0, NULL, GST_GL_TEXTURE_TARGET_2D,
-- 
1.9.1

